src/webgl/p5.Shader.js

```diff
@@ -1326,12 +1326,18 @@ p5.Shader = class {
         }
         break;
       case gl.SAMPLER_2D:
-        gl.activeTexture(gl.TEXTURE0 + uniform.samplerIndex);
-        uniform.texture =
-          data instanceof p5.Texture ? data : this._renderer.getTexture(data);
-        gl.uniform1i(location, uniform.samplerIndex);
-        if (uniform.texture.src.gifProperties) {
-          uniform.texture.src._animateGif(this._renderer._pInst);
+        if (typeof data == 'number') {
```

Comment:
How would you feel about, rather than directly passing in a number, we do something like:
```js
function rawTexture(data) {
  return { type: 'texture', data }
}

// Usage:
myShader.setUniform(rawTexture(myTex))
```

so that we could then check if `data?.type === 'texture'`? The reasoning being, this would distinguish it from numbers that represent actual numbers, so we'd be able to provide friendly errors in the future if a user passes a number into a texture uniform. We could make separate functions like this in the future as well for different types and similarly show friendly errors if e.g. you pass a 2D texture into a cubemap uniform.

Reply:
That wouldn't bind to a texture, but a texture slot, and it creates a new object so by p5.js naming it should start with `create`. That would make it `createTextureSlot()` which doesn't make sense to first viewers whether inexperienced or experienced with WebGL.

Personally, I don't think that it'll be too much of an issue not providing a warning, but if we really want to provide an issue wouldn't it be better to use the WebGL texture slot constants instead of creating objects or try and get the WebGL errors and use that for a warning if we really don't want to use the WebGL texture slot constant for some reason? It's pretty unlikely that anyone's going to accidentally give that as a number.